extension/src/extension.ts

```diff
@@ -67,16 +67,33 @@ export function activate(context: vscode.ExtensionContext) {
     });
 
     const disposableBuild = vscode.commands.registerCommand("xcomponent.build.project", () => {
+        const xcbuildPath = vscode.workspace.getConfiguration()["xcbuild"];
+        if (!fs.existsSync(xcbuildPath)) {
+            vscode.window.showErrorMessage(`xcbuild.exe not found. Please specify xcbuild path in vscode settings`);
+            return;
+        }
         const rootPath = vscode.workspace.rootPath;
         const baseName = path.basename(rootPath);
         const xcmlPath = `${rootPath}${path.sep}${baseName}_Model.xcml`;
-        const buildCommand = `xcbuild --compilationmode=Debug --build --env=Dev --vs=VS2015 --project=${xcmlPath}`;
-        if (fs.existsSync(xcmlPath)) {
+        if (!fs.existsSync(xcmlPath)) {
+            vscode.window.showErrorMessage(`Build error. File ${xcmlPath} not found`);
+            return;
+        }
+        const isWindowsPlatform = /^win/.test(process.platform);
+        if (isWindowsPlatform) {
+            const buildCommand = `${xcbuildPath} --compilationmode=Debug --build --env=Dev --vs=VS2015 --project=${xcmlPath}`;
             terminal.show();
             terminal.sendText(buildCommand);
-        } else {
-            vscode.window.showErrorMessage(`Build error. File ${xcmlPath} not found`);
+            return;
+        }
+        const monoPath = vscode.workspace.getConfiguration()["mono"];
+        if (!fs.existsSync(monoPath)) {
```

Comment:
I think checking if the file exist does not work when mono is in the path env

Reply:
The command to build is the following : 
mono  xcbuild--compilationmode=Release --build --framework=Framework452 --env=Dev --vs=VS2015 --monoPath=“/Library/Frameworks/Mono.framework/Versions/Current//lib/mono/4.5/Facades” --project=“OrderProcessing_Model.xcml”

Reply:
So the command fails if monopath is not specified, even if mono is in the environnement path

Reply:
I think there is a confusion because monoPath in the command line is not the path to the mono exe but a path to mono libraries

Reply:
So as I understand your code, mono has to be in the environnement path and the path lib  defined in the settings

Reply:
You have to explain that in the readme !